Source/gg2/Objects/InGameElements/Character.events/User Event 13.xml

```diff
@@ -54,12 +54,14 @@ if(global.updateType == FULL_UPDATE){
             break;
         case CLASS_SNIPER:
             currentWeapon.t = read_ubyte(global.deserializeBuffer);
+            if(currentWeapon.t) zoomed = true; // Quick hack
```

Comment:
Not really part of your changes but urgh, that variable naming...
On something related to your code though, comments like this are not very informative, and I'd even argue it's not much of a hack at all.